docs/pages/joy.tsx

```diff
@@ -0,0 +1,207 @@
+import * as React from 'react';
+import { CssVarsProvider, styled, useColorScheme } from '@mui/joy/styles';
+import { GlobalStyles } from '@mui/styled-engine';
+
+declare module '@mui/joy/styles' {
+  interface ColorSchemeOverrides {
+    dark: true;
+    lightGreen: true;
+    darkGreen: true;
+    // lightRed: true;
+    // darkRed: true;
+  }
+  interface Palette {
+    bgcolor: React.CSSProperties['backgroundColor'];
+    text: React.CSSProperties['color'];
+  }
+}
+
+const Button = styled('button', {
+  shouldForwardProp: (prop) => prop !== 'active',
+})<{ active?: boolean; color?: string }>(({ theme, active }) => ({
+  border: 0,
+  fontSize: theme.vars.fontSize.md,
+  height: 40,
+  borderRadius: 40,
+  padding: '0.25rem 1rem',
+  ...(active && {
+    color: '#fff',
+    backgroundColor: theme.vars.palette.brand[500],
+  }),
+}));
+
+const Button2 = styled('button', {
+  shouldForwardProp: (prop) => prop !== 'active',
+})<{ active?: boolean; color?: string }>(({ theme, active }) => ({
+  fontSize: theme.vars.fontSize.md,
+  height: 40,
+  borderRadius: 40,
+  padding: '0.25rem 1rem',
+  border: '1px solid',
+  backgroundColor: 'transparent',
+  color: theme.vars.palette.text,
+  opacity: 0.4,
+  ...(active && {
+    borderColor: theme.vars.palette.text,
+    opacity: 1,
+  }),
+}));
+
+const Toggle = () => {
+  const [mounted, setMounted] = React.useState(false);
+  const { mode, setMode, colorScheme, setColorScheme } = useColorScheme();
+
+  // When mounted on client, now we can show the UI
+  React.useEffect(() => setMounted(true), []);
```

Comment:
Why do we need this?

Reply:
To prevent SSR className mismatch because on the server there is no `mode | colorScheme`. Here is the explanation https://github.com/pacocoursey/next-themes#usetheme

> This page is for demo only, will be removed before merging.